torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,328 @@
+import os
+import shutil
+
+from torch import Tensor
+from typing import Callable, Optional, Dict, Tuple
+import rasterio
+import numpy as np
+import torch
+import pandas as pd
+import matplotlib.pyplot as plt
+
+from .geo import NonGeoDataset
+from .utils import check_integrity, extract_archive, download_url
+
+"""MapInWild dataset."""
+
+class MapInWild(NonGeoDataset):
+    """MapInWild dataset.
+
+        The `MapInWild <https://arxiv.org/abs/2212.02265>` is curated for the task of wilderness mapping on a pixel-level. 
+        MapInWild is a multi-modal dataset and comprises various geodata acquired and formed from different RS sensors over 1018 locations:
+        dual-pol Sentinel1, four-season Sentinel-2 with 10 bands, ESA WorldCover map, and Visible Infrared Imaging Radiometer Suite NightTime Day/Night band.
+        The dataset consists of 8144 images with the shape of 1920 × 1920 pixels. 
+        The images are weakly annotated from the World Database of Protected Areas(WDPA).
+
+        Dataset features:
+        * 1018 areas globally sampled from the WDPA 
+        * 10-Band Sentinel-2
+        * Dual-pol Sentinel-1 
+        * ESA WorldCover Land Cover 
+        * Visible Infrared Imaging Radiometer Suite NightTime Day/Night Band
+
+        If you use this dataset in your research, please cite the following paper:
+
+        * https://arxiv.org/abs/2212.02265
+    """
+    BAND_SETS: Dict[str, Tuple[str, ...]] = {
+        "all": (
+            "VV",
+            "VH",
+            "B2",
+            "B3",
+            "B4",
+            "B5",
+            "B6",
+            "B7",
+            "B8",
+            "B8A",
+            "B11",
+            "B12",
+            "2020_Map",
+            "avg_rad"
+            ), 
+        "s1": ("VV", "VH"),
+        "s2": (
+            "B2",
+            "B3",
+            "B4",
+            "B5",
+            "B6",
+            "B7",
+            "B8",
+            "B8A",
+            "B11",
+            "B12"
+        ),
+        "esa_wc":{"2020_Map"},
+        "viirs":{"avg_rad"}
```

Comment:
I'm guessing these should be tuples instead of sets:
```suggestion
        "esa_wc": ("2020_Map",),
        "viirs": ("avg_rad",),
```